### PR TITLE
Feature esp update

### PR DESCRIPTION
**Pull request by PatrickDekker98**
updated code on esp and code interfacing with esp. 
esp_32_c now makes use of the handshake implemented on esp
the spi slave is initialized before socket connection on esp

### CODE REVIEW

File: code/src/esp_32.cpp [hunk 1]
```diff
@@ -23,10 +27,22 @@ namespace r2d2::communication {
 
     bool esp_32_c::receive(r2d2::frame_external_s &frame) {
         if (recv_queue.empty()) {
-            return false;
-        } else {
-            frame = recv_queue.copy_and_pop();
-            return true;
+            // set the first bite of the buffer to 0
+            // only the first because teh fist byte of an external frame is its length
```
Comment (TonyvanderKrogt) [inline code/src/esp_32.cpp]: Spelling error 5th word

**Review by TonyvanderKrogt — CHANGES_REQUESTED** [review]
Everything is fine except for 1 spelling error

File: code/src/esp_32.cpp (hunk 1 above)
Comment (PatrickDekker98) [reply code/src/esp_32.cpp]: fixed in 5ed3bd3

**Review by TonyvanderKrogt — APPROVED** [review]
Thank you for the small change in the comments

**Pull request merged**